### ISSUE TITLE
Use facemesh for filters?

### ISSUE THREAD

**Issue by tjasmin111** [thepowerfuldeez/facemesh.pytorch#8]
Interesting project. Is it possible to use facemesh to change face and make nose smaller? How can I do that?

**Comment by Lirui991225** [comment]
describe more clearly

**Comment by arianaa30** [comment]
I mean, let's say I want to make some changes in the face landmarks like those beauty apps. For instance, to make nose smaller or lips bigger and such. How to achieve that?

**Comment by tjasmin111** [comment]
I mean, let's say I want to make some changes in the face landmarks like those beauty apps. For instance, to make nose smaller or lips bigger and such. How to achieve that?

**Comment by Lirui991225** [comment]
Too easy！

**Comment by arianaa30** [comment]
Hmm can you please explain how? We can "get" the facemesh but how to use that to change the face landmarks?

**Comment by tjasmin111** [comment]
Hmm can you please explain how? We can "get" the facemesh but how to use that to change the face landmarks?

**Comment by Lirui991225** [comment]
My English is poor,do you know chinese?Can we communicate in Chinese？

**Comment by Lirui991225** [comment]
or we can use the wchat

**Comment by tjasmin111** [comment]
我不懂中文。 但你可以用中文解释，我可以使用谷歌翻译。 所以我的问题是，我们可以“获取”面部网格，但如何使用它来改变面部标志？

**Comment by Lirui991225** [comment]
Perhaps you can perform some transformations on the pixels within a certain range around the landemarkes.

**Comment by Lirui991225** [comment]
Why can you replay to me so quickly

**Comment by tjasmin111** [comment]
Hehe because this is important to me. I need to know how to do it.

**Comment by Lirui991225** [comment]
Are you working or studying

**Comment by tjasmin111** [comment]
What kind of transformation? Can you write an example that gets facemesh and transforms nose?

**Comment by tjasmin111** [comment]
I am an intern.

**Comment by Lirui991225** [comment]
Oh, where are you from? Which school are you from

**Comment by Lirui991225** [comment]
me，too

**Comment by tjasmin111** [comment]
Ok! I'm in the US. California.

**Comment by Lirui991225** [comment]
OK!What is your major，my main focus now is on computer vision and AI

**Comment by Lirui991225** [comment]
> What kind of transformation? Can you write an example that gets facemesh and transforms nose
Can't your third stars solve your problem

**Comment by tjasmin111** [comment]
Yes I'm a computer scientist focusing on computer vision. Do you think you can write a sample app to see how we can change landmark? What transformation should we use??

**Comment by Lirui991225** [comment]
OK!I'll try to write it down.

**Comment by tjasmin111** [comment]
Ok thanks. If we can do it for nose and make it smaller, then I can do it for all landmarks.

**Comment by Lirui991225** [comment]
What time is it in your area

**Comment by Lirui991225** [comment]
i think i can use the opencv todo it

**Comment by tjasmin111** [comment]
It's 2:30am! And still awake, googling for this problem. I hope you can do it with opencv.
So after knowing the mesh coordinates, you use opencv for transformation?

**Comment by Lirui991225** [comment]
Some mesh deformation algorithms can be used, such as Laplacian deformation.

**Comment by Lirui991225** [comment]
I need to know your specific task now.

**Comment by tjasmin111** [comment]
Let's make nose smaller. That's the task.

**Comment by tjasmin111** [comment]
How to use laplacian transformation? Should we pass the nose coordinates or the whole face mesh?

**Comment by Lirui991225** [comment]
> Let's make nose smaller. That's the task.
The image or 3Dfile？

**Comment by tjasmin111** [comment]
Prob image? Let's assume we stream from laptop camera.

**Comment by Lirui991225** [comment]
> How to use laplacian transformation? Should we pass the nose coordinates or the whole face mesh?

If your task is 3d object you can do this transformation on the object's triangular plate,and if it is on image i
think can do some another transformation of 2D image.

**Comment by Lirui991225** [comment]
> Prob image? Let's assume we stream from laptop camera.

Image stream of camera？

**Comment by tjasmin111** [comment]
Yeah, because imagine a person is trying to slim nose on his phone camera. Actually, it is 3D mesh of face bc we use facemesh.

**Comment by tjasmin111** [comment]
Can you just make a small puthon code that make our nose look smaller?

**Comment by Lirui991225** [comment]
import cv2
img = cv2.imred('./test.jpg')
nose_mesh = []
new_width,new_height=a,b#a and b are the new width and height of the nose
letf,right,top,bottom=nose_mesh[:,0].min(),nose_mesh[:,0].max(),nose_mesh[:,1].min(),nose_mesh[:,1].max()
nose_np_array = img[top:bottom,left:right,:]
new_nose_array = cv2.resize((new_width,new_height))
#do a img fusion

**Comment by tjasmin111** [comment]
Did you test it? Can you please post How does the before and after image look?

**Comment by Lirui991225** [comment]
use the 'Poisson Image Editing'

**Comment by Lirui991225** [comment]
I didn't test it, just provided my ideas

**Comment by tjasmin111** [comment]
Hmm you just get the area of nose and resize it. But how to stitch it with the face with no artifact? 
Please test it. I don't think it's that simple

**Comment by Lirui991225** [comment]
use some image fusion algorithms

**Comment by Lirui991225** [comment]
Just like  'Poisson Image Editing'.

**Comment by tjasmin111** [comment]
Hmm poisson image editing fills the empty pixels with background image...yeah worth a try

**Comment by Lirui991225** [comment]
Yes，you can have a try，there are some tutorials on Google that you can learn

**Comment by Lirui991225** [comment]
May I ask how much your internship salary is ?

**Comment by Lirui991225** [comment]
Hmm

**Comment by tjasmin111** [comment]
Poisson algorithm needs a mask..not sure how to generate it.
Internship is not much...40$/hr

**Comment by Lirui991225** [comment]
It‘s so cool,mysalary is just10000￥ per month.

**Comment by tjasmin111** [comment]
Country to country is different. If you come to US, you can get $50

**Comment by Lirui991225** [comment]
That's true

**Comment by Lirui991225** [comment]
> Poisson algorithm needs a mask..not sure how to generate it. Internship is not much...40$/hr

The mask can be got  by the nose area of mesh

**Comment by tjasmin111** [comment]
But it shouldn't be a single mask. We have to run it continuously for camera. Face pose can change

> The mask can be got by the nose area of mesh

**Comment by Lirui991225** [comment]
Have you read any books on machine learning before，such as the prml,esl,mlapp and so on.

**Comment by tjasmin111** [comment]
Not really. Not familiar with those. Why?

**Comment by Lirui991225** [comment]
Classic books on machine learning are very helpful for a deeper understanding of algorithms, although it may be a bit difficult.

**Comment by tjasmin111** [comment]
I see. If you could do a simple code for our nose problem and make it work for a face video, it would be great. That works on webcam video on your face
